executor/src/wasm_executor.rs

```diff
@@ -330,7 +372,40 @@ mod tests {
 		calldata.extend_from_slice(sig.as_ref());
 		assert_eq!(
 			WasmExecutor.call(&mut ext, &test_code[..], "test_ed25519_verify", &CallData(calldata)).unwrap(),
-			vec![1]
+			vec![0]
```

Comment:
Why this set to `0`? Is this workaround?

Reply:
0 is the legitimately expected result.

Reply:
Hm, could you help to understand this?

1. I thought this is a positive test, i.e. checks for successful verification.
2. Looking at [test_ed25519_verify](https://github.com/paritytech/polkadot/blob/1344dadd131629e39109ca7506d9f004da69eef4/wasm-runtime/test/src/lib.rs#L29) it returns, casted bool taken from `ed25519_verify` to u8, so if `ed25519_verify` returns `true` then it should return `vec![1u8]` or `vec![0u8]` otherwise.
3. [`ed25519_verify`](https://github.com/paritytech/polkadot/blob/1344dadd131629e39109ca7506d9f004da69eef4/wasm-runtime/std/src/lib.rs#L113) have no documentation which one of boolean values returns in case of successful result, but I almost certain that `true` should be returned for successful verification.

Reply:
ugh - you're right!

Reply:
you're right about point 1, but `ed25519_verify` returns a `u32` not a `bool`, with success being 0 and error conditions (of which there are several types) being non-zero. `test_ed25519_verify` is just a passthrough with a `u8` cast, so we still expect a 0 for success.

Reply:
Hm, but if you follow my link, https://github.com/paritytech/polkadot/blob/1344dadd131629e39109ca7506d9f004da69eef4/wasm-runtime/std/src/lib.rs#L113
‘ed25519_verify’ in fact returbs a ‘bool’ not a u32

it is ‘ext_ed25519_verify’ returns u32